core/status/engine.go

```diff
@@ -111,6 +114,54 @@ func (s *status) collectDiagnostics() map[string][]core.DiagnosticResult {
 	return result
 }
 
+func (s *status) checkHealth(ctx echo.Context) error {
+	checkCtx, cancel := context.WithTimeout(ctx.Request().Context(), 5*time.Second)
+	defer cancel()
+
+	result := s.doCheckHealth(checkCtx)
+	responseCode := 200
+	if result.Status != core.HealthStatusUp {
+		responseCode = 503
+	}
+
+	return ctx.JSON(responseCode, result)
+}
+
+func (s *status) doCheckHealth(ctx context.Context) core.HealthCheckResult {
+	results := make(map[string]core.HealthCheckResult, 0)
+	err := s.system.VisitEnginesE(func(engine core.Engine) error {
+		if err := ctx.Err(); err != nil {
```

Comment:
I put it there anticipating there will be slow(er) health checks in future, which we don't want to block the monitoring application fetching the health metrics. But that's (currently) not a factor since we only check an in-memory certificate. To avoid YAGNI I'll just remove time-outs/contexts altogether for now.